### ISSUE TITLE
Environment marker is not shown

### ISSUE THREAD

**Issue by Zorakie** [XjSv/environment-marker#245]
**Describe the bug**
Environment marker is not always loaded on the page (Firefox) for ServiceNow platform.

ServiceNow is a cloud-based platform for automating IT management workflows. As developers, we are in a need of knowing where we are as usually working with multiple instances per customer and all of them look almost the same, especially with their new UI.

ServiceNow has fairly complex UI, so I am not sure if this is strictly a bug of Environment marker, or the platform's UI is somehow overwriting it. The problem is, sometimes it loads, but mostly not

**To Reproduce**
1) Register an account on developer.service-now.com
2) after registration, log in and on the main page, request a new personal developer instance. In the pop-up, select the latest version (Utah)
3) Wait for your instance to be provisioned and log in by given link/credentials.
4) Skip all the annoying walkthrough and analytics requests.
5) set up your environment marker to match the URL, I am using just <instancename>.service-now.com
Play with it a bit, refresh, close browser, return the next day (keep in mind the personal developer instance go to sleep after some time so the next day, you need to log in to the developer portal and wake it up)

**Expected behavior**
When URL matches, the environment marker is always shown.

**Screenshots**
![obrazek](https://github.com/XjSv/environment-marker/assets/14437576/6367a594-a80c-4ca2-bc7f-4e21e1d6b830)

**Information (please complete the following information):**
 - OS: Windows 11
 - Browser: Firefox Developer Edition
 - Browser Version 115.0b9
 - Extension Version 3.0.0

Also, I have disabled all extensions that could possibly interfere (namely: ServiceNow related addon, TamperMonkey, and Stylish. Other addons not related to my customizations were running as I assume compatibility (adblock, Gramarly and plugins not interfering with page layout, like Containers, tab sorter, etc.)

**Comment by XjSv** [comment]
Thanks for the report, I also use the ServiceNow platform so I am able to test it and debug.

**Comment by XjSv** [comment]
Since the update to Manifest V3 there was an additional permission added. @Zorakie Could you check the permissions tab in addons for "Access your data for all websites" and enable it.

![Permissions](https://github.com/XjSv/environment-marker/assets/107681/4ff50ae1-d102-4fd8-be99-d76e5f55dddc)

**Comment by XjSv** [comment]
According to Mozilla this is a known bug. 

- https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/optional_permissions#host_permissions
- https://bugzilla.mozilla.org/show_bug.cgi?id=1766026

**Comment by Zorakie** [comment]
Hi! thank you for your response. I have checked the permissions and it was enabled by default. Do I understand correctly from your links that this must be fixed by Mozilla? Because the strange thing is, the bug you have linked is like a year old and it is still open, while this issue with the Environment marker is ... let's say a month or less old (truly don't know when it stoped working, but I am pretty sure I was successfully using it like last half a year).
Thanks!

**Comment by XjSv** [comment]
No, I was just wondering if maybe after the update to Manifest V3, that the permissions did not apply correctly. But I can see now that this is not the case. 

I did test it using my workplace ServiceNow account and it seemed to be working correctly. Clearly you still have an issue so I will test it again using a personal ServiceNow account.

For now could you please try something out? If possible can you delete the marker for ServiceNow and recreate it manually? I'm just wondering if it's a data issue. Thanks!

**Comment by Zorakie** [comment]
Hi, unfortunately, I have bad news. I tried to remove all my entries. I tried even reinstalling the plugin itself. I have set my marker again enabled the security setting, enabled favicon marker as well, and visited my instance. Both the colored dot and the ribbon kicked in. So I went doing other stuff, switching between tabs, you know, regular work ... suddenly, I see that the dot and the ribbon are missing again 😞

**Comment by BlackMagicCoding** [comment]
Wow, I really just had a good laugh when I read this issue :laughing: 
A few minutes ago I was just about to raise one myself - a coworker of mine and I have this issue as well since a few weeks... both working with ServiceNow, both using Firefox.
Seems that there is a big overlap with ServiceNow and people who use this extension ^^

My other coworkers (also working with ServiceNow) who use Google Chrome instead of Firefox don't have this problem.
This extension worked very well for me when I started using it at the end of last year, but something changed.

Interestingly enough I am able to temporarily fix the issue by opening the environment marker menu and clicking Disable/Enable.
When reloading existing tabs or creating new ones in the browser window, the markers are drawn correctly,... until I close close Firefox and start it anew.

**Comment by BlackMagicCoding** [comment]
![image](https://github.com/XjSv/environment-marker/assets/57157966/a6a24fbe-7f1a-4a11-b592-46247f9e2415)

**Comment by BlackMagicCoding** [comment]
Just some additional info:
* I tried removing and then reinstalling the extension from scratch
* I tried recreating the environment markers by hand (not importing)
* I checked the permissions

**Comment by XjSv** [comment]
@BlackMagicCoding Thanks for the troubleshooting. I think I have tracked the issue down to the background script stopping for some reason. This started since the Manifest V3 update.

![Background-Script](https://github.com/XjSv/environment-marker/assets/107681/11021a70-0650-4402-8a40-e577be959a4c)

I am still investigating why this is happening.

**Comment by XjSv** [comment]
@BlackMagicCoding & @Zorakie Just wanted to mention that I just pushed out the latest release that contains a hotfix for the background script issue. Thanks to the both of you for reporting this. 

If the issue still exists after v3.0.1, please open up another ticket.

**Comment by BlackMagicCoding** [comment]
No, I have to thank YOU for fixing this so quickly ;)
I just downloaded the updated version of the extension and it seems to work correctly now.

Installing the extension update dropped the permission, which I simply had to set again.
I am not absolutely 100% sure, but I think it also dropped the environment markers as well, which I then simply imported from my backup... could be that was me though, since I did a lot of different things when trying to figure out what is going wrong, so take this with a pinch of salt.
